### PR TITLE
mm/mm_check_heap_corruption.c : Show pid with signed format

### PR DESCRIPTION
**Pull request by jeongchanKim**
For stack allocation, the pid which is in mm_allocnode_s is pid * (-1) for distinguishing it is stack allocation or normal allocation.
If we print the pid with unsigned format, then the pid of stack allocated node looks very big number because of negative value.
So it is better to print the pid with signed format.

Signed-off-by: jeongchanKim <jc_.kim@samsung.com>

### CODE REVIEW

File: os/mm/mm_heap/mm_check_heap_corruption.c [hunk 1]
```diff
@@ -102,12 +102,12 @@ static void dump_node(struct mm_allocnode_s *node, node_type_t type)
 #ifdef CONFIG_DEBUG_MM_HEAPINFO
 #if CONFIG_TASK_NAME_SIZE > 0
 	if (prctl(PR_GET_NAME, myname, node->pid) == OK) {
-		dbg("Node owner pid = %u (%s), allocated by code at addr = 0x%08x\n", node->pid, myname, node->alloc_call_addr);
+		dbg("Node owner pid = %d (%s), allocated by code at addr = 0x%08x\n", node->pid, myname, node->alloc_call_addr);
 	} else {
-		dbg("Node owner pid = %u (Exited Task), allocated by code at addr = 0x%08x\n", node->pid, node->alloc_call_addr);
+		dbg("Node owner pid = %d (Exited Task), allocated by code at addr = 0x%08x\n", node->pid, node->alloc_call_addr);
```
Comment (sunghan-chang) [inline os/mm/mm_heap/mm_check_heap_corruption.c:107]: If node->pid is negative, it's not `Exited Task`. How about splitting negative case for `node->pid`?

Comment (sunghan-chang) [reply os/mm/mm_heap/mm_check_heap_corruption.c:107]: And please consider `prctl` argument for `node->pid` as well for negative.

Comment (jeongchanKim) [reply os/mm/mm_heap/mm_check_heap_corruption.c:114]: Fixed. Thanks.

File: os/mm/mm_heap/mm_check_heap_corruption.c [hunk 2]
```diff
@@ -100,14 +100,21 @@ static void dump_node(struct mm_allocnode_s *node, node_type_t type)
...
+		pid = (-1) * pid;
+	}
 #if CONFIG_TASK_NAME_SIZE > 0
-	if (prctl(PR_GET_NAME, myname, node->pid) == OK) {
-		dbg("Node owner pid = %u (%s), allocated by code at addr = 0x%08x\n", node->pid, myname, node->alloc_call_addr);
+	if (prctl(PR_GET_NAME, myname, pid) == OK) {
+		dbg("Node owner pid = %d (%s), allocated by code at addr = 0x%08x\n", node->pid, myname, node->alloc_call_addr);
 	} else {
-		dbg("Node owner pid = %u (Exited Task), allocated by code at addr = 0x%08x\n", node->pid, node->alloc_call_addr);
+		dbg("Node owner pid = %d (Exited Task), allocated by code at addr = 0x%08x\n", node->pid, node->alloc_call_addr);
```
Comment (sunghan-chang) [inline os/mm/mm_heap/mm_check_heap_corruption.c:114]: How about this?
```
	pid_t pid = node->pid;
	char is_stack[9] = "'s stack";

	if (pid < 0) {
		/* For stack allocated node, pid is negative value.
		 * To use the pid, change it to original positive value.
		 */
		pid = (-1) * pid;
	} else {
		is_stack[0] = '\n';
	}

#if CONFIG_TASK_NAME_SIZE > 0
	if (prctl(PR_GET_NAME, myname, pid) == OK) {
		dbg("Node owner pid = %d (%s%s), allocated by code at addr = 0x%08x\n", node->pid, myname, is_stack, node->alloc_call_addr);
	} else {
		dbg("Node owner pid = %d (Exited Task%s), allocated by code at addr = 0x%08x\n", node->pid, is_stack, node->alloc_call_addr);
```

Comment (jeongchanKim) [reply os/mm/mm_heap/mm_check_heap_corruption.c]: Modified as you suggested. Thanks.

**Review by thapav — APPROVED** [review]
LGTM

File: os/mm/mm_heap/mm_check_heap_corruption.c [hunk 3]
```diff
@@ -100,14 +101,23 @@ static void dump_node(struct mm_allocnode_s *node, node_type_t type)
...
-		dbg("Node owner pid = %u (%s), allocated by code at addr = 0x%08x\n", node->pid, myname, node->alloc_call_addr);
+	if (prctl(PR_GET_NAME, myname, pid) == OK) {
+		dbg("Node owner pid = %d (%s%s), allocated by code at addr = 0x%08x\n", node->pid, myname, is_stack, node->alloc_call_addr);
 	} else {
-		dbg("Node owner pid = %u (Exited Task), allocated by code at addr = 0x%08x\n", node->pid, node->alloc_call_addr);
+		dbg("Node owner pid = %d (Exited Task%s), allocated by code at addr = 0x%08x\n", node->pid, is_stack, node->alloc_call_addr);
 	}
 #else
-	dbg("Node owner pid = %u, allocated by code at addr = 0x%08x\n", node->pid, node->alloc_call_addr);
+	dbg("Node owner pid = %d, allocated by code at addr = 0x%08x\n", node->pid, node->alloc_call_addr);
```
Comment (sunghan-chang) [inline os/mm/mm_heap/mm_check_heap_corruption.c:120]: How about adding "stack" here as well?
dbg("Node owner pid = %d%s, allocated by code at addr = 0x%08x\n", node->pid, is_stack, node->alloc_call_addr);

Comment (jeongchanKim) [reply os/mm/mm_heap/mm_check_heap_corruption.c]: Added. Thanks.

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**